File: DSA-Sheets/Striver-SDE-Sheet/02-Arr-II/Rotate Matrix by 90 deg/brute-force.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        n = len(matrix) # getting #rows 
        rotated = [[0 for _ in range(n)] for _ in range(n)] # make list of list
        for i in range(n):
            for j in range(n):
                rotated[j][n-i-1] = matrix[i][j]
        return rotated

File: DSA-Sheets/Striver-SDE-Sheet/02-Arr-II/Rotate Matrix by 90 deg/brute-force.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    void rotate(vector<vector<int>> &matrix)
    {
        int n = matrix.size();
        vector<vector<int>> rotated(n, vector<int>(n, 0));
        // simply map row to coloumn
        for (int i = 0; i < n; i++)
            for (int j = 0; j < n; j++)
                rotated[j][n - i - 1] = matrix[i][j];
        matrix = rotated;
    }
};

// time  O(n2)
// space  O(n2)